beacon_node/http_api/src/lib.rs

```diff
@@ -783,39 +783,112 @@ pub fn serve<T: BeaconChainTypes>(
+                                } else {
+                                    None
+                                };
+
+                                let committee_cache = match maybe_cached_shuffling {
+                                    Some(shuffling) => shuffling,
+                                    None => {
+                                        let possibly_built_cache =
+                                            match RelativeEpoch::from_epoch(current_epoch, epoch) {
+                                                Ok(relative_epoch)
+                                                    if state.committee_cache_is_initialized(
+                                                        relative_epoch,
+                                                    ) =>
+                                                {
+                                                    state
+                                                        .committee_cache(relative_epoch)
+                                                        .map(Cow::Borrowed)
+                                                }
+                                                _ => CommitteeCache::initialized(
+                                                    state,
+                                                    epoch,
+                                                    &chain.spec,
+                                                )
+                                                .map(Cow::Owned),
+                                            }
+                                            .map_err(
+                                                |e| match e {
+                                                    BeaconStateError::EpochOutOfBounds => {
+                                                        let max_sprp =
+                                                            T::EthSpec::slots_per_historical_root()
+                                                                as u64;
+                                                        let first_subsequent_restore_point_slot =
+                                                            ((epoch.start_slot(
+                                                                T::EthSpec::slots_per_epoch(),
+                                                            ) / max_sprp)
+                                                                + 1)
+                                                                * max_sprp;
+                                                        if epoch < current_epoch {
+                                                            warp_utils::reject::custom_bad_request(
+                                                                format!(
                                                     "epoch out of bounds, try state at slot {}",
                                                     first_subsequent_restore_point_slot,
```

Comment:
the indentation on this is messed up, likely because of the right-ward drift

Rustfmt is probably freaking out because the format string goes over the 100 column limit. If we un-indent one level it might fix it, or we could put a line break in the string like:

```rust
"epoch out of bounds. \
 try state at slot {}"
```